_posts/2015-07-21-the-epages-selenium-framework.md

```diff
@@ -0,0 +1,138 @@
+internally you can read a description of its architecture 
+[here][selenium-architecture].
+
+# Basic concepts of the ESF
+
+In order to make it as easy as possible for our developers to write
+automated UI tests, and in order to avoid duplicated code from
+developing in different teams, we have built a framework around
+Selenium: the ePages Selenium Framework or just ESF.
+
+The ESF is a Java project build upon Selenium WebDriver
+and the test framework [TestNG]. It uses [Gradle] as build tool and
+supports starting tests from within an IDE (e.g. Eclipse) and from
+the command-line. Browser start and test report generation is
+controlled with the help of TestNG configuration methods and event listeners. 
+
+The actual test suite consists of test scripts and Page Objects.
+Page Objects are abstractions of individual web pages. They provide
+service methods that enable a user to interact with the
+underlying web page. These service methods encapsulate the
+locators for the web elements and all of the logic to gather
+information from the page or interacting with it. The Tests
+are then using these service methods to fulfill defined
+test specifications. The following picture shows this structure.
+
+{% image esf-concept.png %}{% endimage %}
+
+Here you can see a short test script for the search functionality.
+The test scripts are always derived from the base class `TestBaseEp6`.
+That way the developer can focus on writing the test method, leaving
+all the configuration and setup work to the framework. One of the
+first steps of each test script is to
+get an instance of the `ActionBot`. This class provides access to
+the `WebDriver` and additional utility methods like navigating through
+the ePages shop. This particular test will go to an ePages storefront,
+search for a product with the name "Mag-Lite Mini" and check if the
+product is part of the search result.
+
+    [...]
+    
+    public final class SearchTest extends TestBaseEp6 {
+    
+        @Test(groups = { "SEARCH", "CORE” })
+        public void simpleSearchTest() {
+    
+            final String productName = "Mag-Lite Mini";
+    
+            final ActionBotEp6 actionBot = this.getActionBot();
+            final PageHomepage sfLanding = actionBot.goToStorefront();
+            final PageSearchResults searchResults = sfLanding.searchFor(productName);
+    
+            assertTrue("The product we searched for is part of the search result.",
+                    searchResults.isProductPartOfSearchResult(productName));
+        }
+    }
+
+# Role in the release process
+
+The ESF plays an important role in the ePages release process.
+As we are releasing a new ePages version every two weeks, it is impossible to
+do manual regression testing for all our features, more than fifty technology partners,
+various installation types
+and support for different Linux distributions. Beside a lot of other testing, like unit and
```

Comment:
it looks like this is on a new sentence rather than a continuation from the previous